src/cascadia/TerminalApp/TerminalSettings.cpp

```diff
@@ -6,8 +6,195 @@
+        if (profile.Background())
+        {
+            _DefaultBackground = til::color{ profile.Background().Value() };
+        }
+        if (profile.SelectionBackground())
+        {
+            _SelectionBackground = til::color{ profile.SelectionBackground().Value() };
+        }
+        if (profile.CursorColor())
+        {
+            _CursorColor = til::color{ profile.CursorColor().Value() };
+        }
+
+        _ScrollState = profile.ScrollState();
+
+        if (!profile.BackgroundImagePath().empty())
+        {
+            _BackgroundImage = profile.GetExpandedBackgroundImagePath();
+        }
+
+        _BackgroundImageOpacity = profile.BackgroundImageOpacity();
+        _BackgroundImageStretchMode = profile.BackgroundImageStretchMode();
+
+        _BackgroundImageHorizontalAlignment = profile.BackgroundImageHorizontalAlignment();
+        _BackgroundImageVerticalAlignment = profile.BackgroundImageVerticalAlignment();
+
+        _RetroTerminalEffect = profile.RetroTerminalEffect();
+
+        _AntialiasingMode = profile.AntialiasingMode();
+
+        if (profile.TabColor())
+        {
+            const til::color colorRef{ profile.TabColor().Value() };
+            _TabColor = static_cast<uint32_t>(colorRef);
+        }
+    }
+
+    // Method Description:
+    // - Applies appropriate settings from the globals into the TerminalSettings object.
+    // Arguments:
+    // - globalSettings: the global property values we're applying.
+    // Return Value:
+    // - <none>
+    void TerminalSettings::_ApplyGlobalSettings(const TerminalApp::GlobalAppSettings& globalSettings) noexcept
+    {
+        _InitialRows = globalSettings.InitialRows();
+        _InitialCols = globalSettings.InitialCols();
+
+        _WordDelimiters = globalSettings.WordDelimiters();
+        _CopyOnSelect = globalSettings.CopyOnSelect();
+        _ForceFullRepaintRendering = globalSettings.ForceFullRepaintRendering();
+        _SoftwareRendering = globalSettings.SoftwareRendering();
+        _ForceVTInput = globalSettings.ForceVTInput();
+    }
+
+    // Method Description:
+    // - Apply a given ColorScheme's values to the TerminalSettings object.
+    //      Sets the foreground, background, and color table of the settings object.
+    // Arguments:
+    // - scheme: the ColorScheme we are applying to the TerminalSettings object
+    // Return Value:
+    // - <none>
+    void TerminalSettings::ApplyColorScheme(const TerminalApp::ColorScheme& scheme)
+    {
+        _DefaultForeground = til::color{ scheme.Foreground() };
+        _DefaultBackground = til::color{ scheme.Background() };
+        _SelectionBackground = til::color{ scheme.SelectionBackground() };
+        _CursorColor = til::color{ scheme.CursorColor() };
+
+        const auto table = scheme.Table();
+        auto const tableCount = gsl::narrow_cast<int>(table.size());
+        for (int i = 0; i < tableCount; i++)
+        {
+            SetColorTableEntry(i, til::color{ table[i] });

```

Comment:
since this is all internal now...
can you just

```c++
std::transform(table.cbegin(), table.cend(), _colorTable.begin(), [](auto&& color) {
    return static_cast<uint32_t>(til::color{ entry });
});
```

and delete `SetColorTableEntry` from the idl and from the public interface and the private one and everything.